### PR TITLE
feat: support mvn --scan-all-unmanaged

### PR DESCRIPTION
**Pull request by orsagie**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Add support for scanning multiple jars under the same folder with `--scan-all-unmanaged` flag.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,9 +27,11 @@
     "find-circular": "npm run build && madge --circular ./dist",
     "format": "prettier --write '{src,test,scripts}/**/*.{js,ts}'",
     "prepare": "npm run build",
-    "tap": "tap test/*.test.* test/acceptance/**/*.test.* test/system/*.test.* -Rspec --timeout=300 --node-arg=-r --node-arg=ts-node/register",
-    "test": "npm run test-common && npm run tap",
-    "test-common": "npm run check-tests && npm run build && npm run lint && node --require ts-node/register src/cli test --org=snyk",
+    "test:common": "npm run check-tests && npm run build && npm run lint && node --require ts-node/register src/cli test --org=snyk",
```
Comment (lili2311) [inline package.json]: what changed here? hard to read it :)

File: src/lib/snyk-test/run-test.ts [hunk 2]
```diff
@@ -257,7 +257,10 @@ async function getDepsFromPlugin(
   root,
   options: Options,
 ): Promise<pluginApi.MultiProjectResult> {
-  options.file = options.file || detect.detectPackageFile(root);
+  // don't override options.file if scanning multiple files at once
+  if (!options.scanAllUnmanaged) {
```
Comment (lili2311) [inline src/lib/snyk-test/run-test.ts]: what happens if you do:
1.  `snyk test a b --scan-all-unmanaged`?
2. `snyk test --file=manifest --scan-all-unmanaged`?

Comment (orsagie) [reply src/lib/snyk-test/run-test.ts]: The real question is what happens on `monitor`
1. `snyk monitor a b --scan-all-unmanaged` will create a project for all unmanaged under `a`, then a project for all unmanaged under `b`
2. `snyk monitor --file=manifest --scan-all-unmanaged` will prioritize the `--scan-all-unmanaged` flag, but will also still have a file attached, which makes no sense. 

I will add a check to say these two flags don't play nice together.

File: package.json (hunk 1 above)
Comment (orsagie) [reply package.json]: Split up into acceptance, system, common, test, based on their path. So it's easier to run a single suite at a time.

File: src/lib/snyk-test/run-test.ts (hunk 2 above)
Comment (lili2311) [reply src/lib/snyk-test/run-test.ts]: 👌

File: src/cli/index.ts [hunk 3]
```diff
@@ -149,6 +149,9 @@ async function main() {
   let failed = false;
   let exitCode = EXIT_CODES.ERROR;
   try {
+    if (args.options.scanAllUnmanaged && args.options.file) {
```
Comment (orsagie) [inline src/cli/index.ts]: @lili2311 check for unsupported options combo

Comment (lili2311) [reply src/cli/index.ts]: thanks, is there a fixture repo I can test this all on?

**Pull request merged**